Godeps/_workspace/src/github.com/whyrusleeping/multiaddr-filter/mask.go

```diff
@@ -0,0 +1,19 @@
+package mask
+
+import (
+	"errors"
+	"net"
+	"strings"
+)
+
+func NewMask(a string) (*net.IPNet, error) {
+	parts := strings.Split(a, "/")
+	if len(parts) == 5 && parts[1] == "ip4" && parts[3] == "ipcidr" {
+		_, ipn, err := net.ParseCIDR(parts[2] + "/" + parts[4])
+		if err != nil {
+			return nil, err
+		}
+		return ipn, nil
+	}
+	return nil, errors.New("invalid format")
+}
```

Comment:
it would be nice if there was an `AddProtocol` method that extension libs could call in an init function. So importing my lib would add my libs protocol to multiaddrs parsing table.